control-plane/p4RuntimeSerializer.cpp

```diff
@@ -639,6 +640,84 @@ getMatchType(cstring matchTypeName) {
     }
 }
 
+/*
+ * This function supports a deeply nested mix of Type_Newtype and typedef.
+ * The function also supported Type_Bits.
+ * A non-zero width, or zero width, or -1 is returned by midWidthbits() used
+ * by this function. For -1, a failure code, the BUG_CHECK is invoked.
+ * Therefore the uint32_t returned by the function is fine.
+*/
+static uint32_t
+getTypeWidth(const IR::Type* type, TypeMap* typeMap) {
+    uint32_t w = 0;
+    if (type == nullptr)
+        return w;
+    BUG_CHECK(!type->is<IR::Type_Error>(), "p4runtime does not support type: %1%",
```

Comment:
Several days back, I made this comment to this PR for such a change.

https://github.com/p4lang/p4c/pull/1736#issuecomment-470313237

The compiler crashed 'Compiler Bug: Exiting with SIGSEGV`.  The line of code is

https://github.com/p4lang/p4c/blob/master/frontends/p4/typeMap.cpp#L336

This is why I added this BUG_CHECK in p4runtime generation to catch more gracefully.

Reply:
The compiler crashes with a SEGV because `errorPosition` is null in `TypeMap::minWidthBits`.  

The following code change fixes the crash.

```
@@ -338,16 +333,14 @@ int TypeMap::minWidthBits(const IR::Type* type, const IR::Node* errorPosition) {
         return 0;
     }
 
-    ::error("%1%: width not well-defined", errorPosition);
+    if (errorPosition) 
+        ::error("%1%: width not well-defined", errorPosition);
+    else
+        ::error("width not well-defined");
     return -1;
 }
```